static/js/components/CandidatesPreferences.jsx

```diff
@@ -0,0 +1,564 @@
+    },
+    "& div": {
+      width: "100%",
+    },
+  },
+}));
+
+function getStyles(classification, selectedClassifications, theme) {
+  return {
+    fontWeight:
+      selectedClassifications.indexOf(classification) === -1
+        ? theme.typography.fontWeightRegular
+        : theme.typography.fontWeightMedium,
+  };
+}
+
+const CandidatesPreferences = () => {
+  const preferences = useSelector((state) => state.profile.preferences);
+  const availableAnnotationsInfo = useSelector(
+    (state) => state.candidates.annotationsInfo
+  );
+  const classes = useStyles();
+  const theme = useTheme();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    // Grab the available annotation fields for filtering
+    dispatch(candidatesActions.fetchAnnotationsInfo());
+  }, [dispatch]);
+
+  const userAccessibleGroups = useSelector(
+    (state) => state.groups.userAccessible
+  );
+
+  const ITEM_HEIGHT = 48;
+  const MenuProps = {
+    PaperProps: {
+      style: {
+        maxHeight: ITEM_HEIGHT * 4.5,
+        width: 250,
+      },
+    },
+  };
+
+  // Get unique classification names, in alphabetical order
+  const { taxonomyList } = useSelector((state) => state.taxonomies);
+  const latestTaxonomyList = taxonomyList.filter((t) => t.isLatest);
+  let classifications = [];
+  latestTaxonomyList.forEach((taxonomy) => {
+    const currentClasses = allowedClasses(taxonomy.hierarchy).map(
+      (option) => option.class
+    );
+    classifications = classifications.concat(currentClasses);
+  });
+  classifications = Array.from(new Set(classifications)).sort();
+
+  const [selectedClassifications, setSelectedClassifications] = useState([]);
+  const [selectedAnnotationOrigin, setSelectedAnnotationOrigin] = useState();
+
+  const { handleSubmit, getValues, control, errors, reset } = useForm();
+
+  useEffect(() => {
+    reset({
+      groupIDs: Array(userAccessibleGroups.length).fill(false),
+    });
+    // Don't want to reset everytime the component rerenders and
+    // the defaultStartDate is updated, so ignore ESLint here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [reset, userAccessibleGroups]);
+
+  // Set initial form values in the redux state
+  useEffect(() => {
+    dispatch(
+      candidatesActions.setFilterFormData({
+        savedStatus: "all",
+      })
+    );
+    // Don't want to reset everytime the component rerenders and
+    // the defaultStartDate is updated, so ignore ESLint here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch]);
+
+  let formState = getValues({ nest: true });
+
+  const validateGroups = () => {
+    formState = getValues({ nest: true });
+    return formState.groupIDs.filter((value) => Boolean(value)).length >= 1;
+  };
+
+  const validateRedshifts = () => {
+    formState = getValues({ nest: true });
+    return (
+      // Both empty
+      (formState.redshiftMinimum === "" && formState.redshiftMaximum === "") ||
+      // Or both filled out
```

Comment:
Why not allow a min or max alone?

Reply:
The API specifies a `redshiftRange` parameter which specifies both so I just changed the front-end form to reflect that. Before, the form would just ignore the form data if only one was specified. But given that we do handle them separately for sources filtering it probably makes more sense to update the candidates handler and then allow one or the other on the form here.

Reply:
Ah yeah, I forgot the API requires both. If you wanna update that here, that'll save needing to update all of this front-end logic later when we change the API. Thanks